src/classes/Collision/SAT.js

```diff
@@ -0,0 +1,187 @@
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectA
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectB
+     * @returns {Goblin.Collision.SAT.Projection|null}
+     */
+    performSat: ( function() {
+        var negatedNormal = new Goblin.Vector3();
+
+        return function( objectA, objectB ) {
+            var allNormals = [];
+            var normalsA = objectA.faceNormals;
+            var normalsB = objectB.faceNormals;
+            if ( normalsA.length === 0 ) {
+                allNormals = normalsB;
+            } else if ( normalsB === 0 ) {
+                allNormals = normalsA;
+            } else {
+                var crossNormals = this._getAllCrossProductPairs( normalsA, normalsB );
+                allNormals.push.apply( allNormals, normalsA );
+                allNormals.push.apply( allNormals, normalsB );
+                allNormals.push.apply( allNormals, crossNormals );
+                this.removeDuplicatedVectors( allNormals );
+            }
+
+            var allProjections = [];
+            var minimumProjection = null;
+            var minimumOverlap = Infinity;
+
+            for ( var i = 0; i < allNormals.length; i++ ) {
+                var normal = allNormals[ i ];
+                negatedNormal.scaleVector( normal, -1 );
+                var computedNegativeProjection = this._findProjectionWithNormal( allProjections, negatedNormal );
+
+                var projection = null;
+                if ( computedNegativeProjection ) {
+                    projection = computedNegativeProjection.cloneAndNegate();
+                }
+                if ( !projection ) {
+                    var a1 = this._findMaxDotInDirection( objectA, normal );
+                    var a2 = -this._findMaxDotInDirection( objectA, negatedNormal );
+                    var b1 = this._findMaxDotInDirection( objectB, normal );
+                    var b2 = -this._findMaxDotInDirection( objectB, negatedNormal );
+                    projection = new Goblin.Collision.SAT.Projection( normal, a1, a2, b1, b2 );
+                }
+
+                if ( projection.overlap < Goblin.EPSILON ) {
+                    return null;
+                }
+
+                allProjections.push( projection );
+                if ( projection.overlap < minimumOverlap ) {
+                    minimumOverlap = projection.overlap;
+                    minimumProjection = projection;
+                }
+            }
+
+            return minimumProjection;
+        };
+    } )(),
+
+    /**
+     * Removes all duplicates from array of vectors.
+     * @param {Array<Goblin.Vector3>} vectors
+     */
+    removeDuplicatedVectors: function( vectors ) {
+        for ( var i = 0; i < vectors.length; i++ ) {
+            var vector = vectors[ i ];
+            var isDuplicate = false;
+
+            for ( var j = i + 1; j < vectors.length; j++ ) {
+                isDuplicate = vector.equals( vectors[ j ] );
```

Comment:
Is this comparison epsilon-biased?

Reply:
Yep, it's was actually added in this pull request